pyproject.toml

```diff
@@ -28,7 +28,9 @@ dev-dependencies = [
     "pyright>=1.1.385",
     "pytest-benchmark>=4.0.0",
     "pytest>=7.4.0",
+    "pyright>=1.1.403",
```

Comment:
I think you want https://github.com/detachhead/basedpyright according to the internet